implementation/src/test/java/io/smallrye/mutiny/operators/MultiGroupTest.java

```diff
@@ -885,4 +885,28 @@ public void testGroupByWithUpstreamFailure() {
         s2.assertFailedWith(TestException.class, "boom");
         subscriber.assertFailedWith(TestException.class, "boom");
     }
+
+    @Test
+    public void testGroupIntoListsWithMaximumDelay() {
+        Multi<Long> publisher = Multi.createFrom().ticks().every(Duration.ofMillis(10));
+
+        AssertSubscriber<List<Long>> subscriber = publisher.groupItems().intoLists().of(3, Duration.ofMillis(100))
+                .subscribe()
+                .withSubscriber(AssertSubscriber.create(10));
+
+        await().until(() -> subscriber.getItems().size() >= 9);
```

Comment:
It won't likely ever happen, but I'd add a timeout guard to the awaitability condition, just to make sure the test cannot get stuck forever.